### PR TITLE
feat: Plain chat

### PR DESCRIPTION
**Pull request by nizzyabi**
## What does this PR do?

This PR replaces the main intercom chat with plain.com's live chat feature.

## Mandatory Tasks (DO NOT REMOVE)

- [Y ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ Y] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ Y] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
     PLAIN_CHAT_ID=
     PLAIN_CHAT_HMAC_SECRET_KEY=
- What are the minimal test data to have?
     Post requests to plain.com
- What is expected (happy path) to have (input and output)?
     Able to see user messages in plain
- Any other important info that could help to test that PR

### CODE REVIEW

**Comment by hbjORbj** [comment]
@nizzyabi Type-check test is failing! You can check https://github.com/calcom/cal.com/actions/runs/12439539908/job/34733552858?pr=18284

File: apps/web/lib/plain/plainChat.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        try {
+          Plain.init(${config ? JSON.stringify(config) : null});
+        } catch (error) {
+          console.error("Failed to initialize Plain:", error);
+        }
+      }
+    }
+  `;
+
+  useEffect(() => {
```
Comment (emrysal) [inline apps/web/lib/plain/plainChat.tsx:236]: This duplicates the hook logic; what do you say about moving this into the hook entirely?

File: apps/web/lib/plain/plainChat.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
+import { useSession } from "next-auth/react";
```
Comment (emrysal) [inline apps/web/lib/plain/plainChat.tsx:1]:
```suggestion
"use client";

import { useSession } from "next-auth/react";
```

File: apps/web/lib/plain/plainChat.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+};
+
+const PlainChat = () => {
+  const [config, setConfig] = useState<PlainChatConfig | null>(null);
+  const { data: session } = useSession();
+  const { openPlain } = usePlain();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
```
Comment (emrysal) [inline apps/web/lib/plain/plainChat.tsx:94]: Not sure this useEffect is needed

**Review by emrysal — APPROVED** [review]
lgtm

**Pull request merged**